### PR TITLE
feat(backend): add db update-naming-conventions command to tracimcli

### PR DESCRIPTION
**Pull request by grignards**
See #3343.

## Checkpoints

These points must be checked before merging. Please don't edit them out.

**For developers**

- [X] If relevant, manual tests have been done to ensure the stability of the whole application and that the involved feature works
- [X] The original issue is up to date w.r.t the latest discussions and contains a short summary of the implemented solution
- [X] Automated tests are irrelevant as this is a command for selected databases coming from v1 version.

**For code reviewers**

- [x] The code is clear enough
- [x] If there are FIXMEs in the code, related issues are mentioned in the FIXME
- [x] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present

**For testers**

- [x] Manual, quality tests have been done

### CODE REVIEW

File: backend/tracim_backend/command/database.py [hunk 1]
```diff
@@ -137,3 +139,75 @@ def take_action(self, parsed_args: argparse.Namespace) -> None:
...
+        "pk": "pk__%(table_name)s",  # Primary keys
+    }
+
+    def get_description(self) -> str:
+        return "Update database naming conventions from V1 database to V2"
+
+    def take_action(self, parsed_args: argparse.Namespace) -> None:
+        super(UpdateNamingConventionsV1ToV2Command, self).take_action(parsed_args)
+        config_uri = parsed_args.config_file
+        # setup_logging(config_uri)
```
Comment (inkhey) [inline backend/tracim_backend/command/database.py:160]: remove this commented line.

**Comment by lebouquetin** [comment]
@grignards @inkhey one small remark about the naming convention you use in the database model.

- Historically, there were names like `fk__content_revisions__owner_id"`
- now there are names like `fk_revision_read_status_revision_id_content_revisions`

It's almost unreadable with no explicit separator, except if you already know all tables and columns. I'd prefer you to use naming convention with explicit separators between field names.

Do you agree?

**Comment by grignards** [comment]
> Do you agree?

In principle yes, I would just be curious to know why those conventions changed as there could be a valid reason.
Also there is a limitation in sqlalchemy when referring multiple columns (with `%(column_0_N_name)s`: either you can have no separator or a `_`, but you cannot tell it to put `__` to separate column names.

**Comment by inkhey** [comment]
> @grignards @inkhey one small remark about the naming convention you use in the database model.
> 
>     * Historically, there were names like `fk__content_revisions__owner_id"`
> 
>     * now there are names like `fk_revision_read_status_revision_id_content_revisions`
> 
> 
> It's almost unreadable with no explicit separator, except if you already know all tables and columns. I'd prefer you to use naming convention with explicit separators between field names.
> 
> Do you agree?

I agree the previous convention is more readable. I do think we should find a way to make thing clear later, but first we should have fixed same rule everywhere. It will make migration way more easier (it may be tricky for sqlite and mariadb though)

**Comment by inkhey** [comment]
> > Do you agree?
> 
> In principle yes, I would just be curious to know why those conventions changed as there could be a valid reason.
> Also there is a limitation in sqlalchemy when referring multiple columns (with `%(column_0_N_name)s`: either you can have no separator or a `_`, but you cannot tell it to put `__` to separate column names.

I don't think there is a good reason. I do not take much attention to database convention when beginning work on tracim2, and i do just don't reuse old tracim v1 metadata config.

**Comment by inkhey** [comment]
verified on branch with old database dump.

**Pull request merged**